### ISSUE TITLE
Member joining a community is asked questions pending approval

### ISSUE THREAD

**Issue by mhulet** [opencollective/community#4]
**Is your feature request related to a problem? Please describe.**
Some communities will want to ask specific questions to a member joining them. It might be to learn more about them, to agree with shared values, etc. Once the request received, Community Links will review answers and approve or reject the request.

Facebook Groups has this feature to screen potential members, and ensure they are the right fit for the group, not just spam or troll.

**Describe the solution you'd like**
When clicking on the new _'Join Community'_ button, a popup opens and ask a few questions with a message explaining:

* why these questions are being asked
* that their request will be pending approval
* that they will be notified by email once Community Links have processed their request

Questions are managed by Community Links in a new _Onboarding_ section of the community settings. What kind of questions we can ask (free text, radio buttons/checkboxes, selects, etc) must still be defined and can be discussed here.

**Describe alternatives you've considered**
Asking questions by email/phone and removing members from a community when required.

**Additional context**
Here is a screenshot showing the new _Onboarding_ section available from the settings menu.

![Screenshot 2020-03-05 at 12 53 02](https://user-images.githubusercontent.com/293355/75979284-719cad00-5ee0-11ea-8d30-3ab49ca52565.png)

**Comment by piamancini** [comment]
@mhulet hi! Quick question, the image you are showing is in the collective's admin panel already, this means the member in question would already be a full admin of the collective to access it. Seems risky if you are screening people to already give them admin rights. 

Some ideas: 
Are these questions public? Could you use the conversations feature to post the questions? Seems like an easy solution. 

If not, another option is to have a custom email to be sent to new members as they are added. Problems with this: 
- We still don't have the feature developed - it's in the potential roadmap for the next quarter but not decided yet. 
- We have a problem with spamming we are tying to solve and one of the options we are considering is sending emails from a 'nonreply' email address which means that the members in question should send the reply somewhere else.

**Comment by mhulet** [comment]
@piamancini Thanks a lot for your comment.

> Quick question, the image you are showing is in the collective's admin panel already, this means the member in question would already be a full admin of the collective to access it. Seems risky if you are screening people to already give them admin rights.

Sorry, my bad. I will update the issue to make it clearer, but this screen is one that will be shown to admins only, sure. I should add a screenshot of the popup we want to show to users.

> Are these questions public? Could you use the conversations feature to post the questions? Seems like an easy solution.

Yes these questions will be public. Anyone who clicks on _'Join Community'_ would see the questions. So yes, in a way, we could use conversations, but I have the feeling that this could work for small collectives, not for bigger ones. And movements like Extinction Rebellion might typically want to add a mandatory checkbox like _"I agree with Extinction Rebellion principles and values"_, something that could be done here and reduce workload for people onboarding new members.

> another option is to have a custom email to be sent to new members as they are added.

Yup, I know that some XR local groups do this, manually. They send a link to a form with a few questions, making it easier for them to onboard new members.

**Comment by mhulet** [comment]
Something else, just thinking about it: a member joining a subcollective should be asked questions from parents collective too.

A typical problem Extinction Rebellion local groups have is that they would like to change the registration form available from the national website, because they have specific questions they want to ask.

For a XR Belgium local group (a subcollective of XR Belgium collective), it would make sense to ask questions asked by XR Belgium AND questions asked by the local group.

**Comment by piamancini** [comment]
> And movements like Extinction Rebellion might typically want to add a mandatory checkbox like "I agree with Extinction Rebellion principles and values", something that could be done here and reduce workload for people onboarding new members.

This can be solved at the host level, if XR has its own host then this is a check like the TOS of the platform. You can add this as Terms of Fiscal Sponsorship. 

My suggestion would be to use conversations for this at the moment and if we see it's growing too much we can evaluate another solution. 

> I should add a screenshot of the popup we want to show to users.

Anything that we do on any admin panel requires you to give permissions to that user in order for them to see it and it seems to defeat the purpose of the screening. 

> Something else, just thinking about it: a member joining a subcollective should be asked questions from parents collective too.

Again, conversations seem to be a flexible enough tool to ask conversations related to both parent and child collective. I honestly don't foresee us handling this complexity in any MVP we could do for this feature.

**Comment by xdamman** [comment]
I’d keep hosts out of that picture. E.g. https://opencollective.com/allforclimate is hosting XR chapters and FFF groups. 

But I could see a screen like we do on mobile when we install a new app that walks you through a few things you need to know:

**Screen 1:**
Welcome to the XR movement!

Pick your identity: ___________________________
*(In the same way that you can pick your identity when contributing to a collective)*

Our demands:
- Tell the Truth
- Act Now
- Power to the people (through Citizen Assemblies)

[ ] I agree

Next ->

**Screen 2:**

Our principles:

- [ ] Principle 1
- [ ] Principle 2
- [ ] Principle 3
- [ ] Principle 4
- [ ] Principle 5

Next ->

**Screen 3**
Tell us more about you:

- Are you willing to be arrested? Yes / Not yet / No
- ...
- ...

Next ->

**Screen 4** (if sub collective)
XR Belgium Legal questions:

Are you a lawyer? Y/N

[Join!]
*Your application will be posted as [your identity] on https://opencollective.com/xr-belgium/conversations/registrations*

—

I like the idea of using conversations because in a way, everybody should be able to see what people have answered. A request to join a collective would effectively be like submitting an expense (for €0). It’s public, anyone can comment on it, and then only of the admins can approve it.

Also to be clear, this is part of OC Community which will be a fork of the front end of open collective. So that we can move swiftly with this and experiment with XR and FFF that need this the most. It would still use the opencollective.com API. If we need to make some changes to it, those will be suggested and/or we could also use a separate database just for the extra info that we would want to also record.

**Comment by mhulet** [comment]
It might be interesting to keep some answers private, visible to admins only.

**Comment by xdamman** [comment]
Yes, in the same way that when you submit an expense, some fields are public and others are private (paypal email, private instructions).
